python_modules/dagster/dagster/_core/execution/job_backfill.py

```diff
@@ -212,7 +252,51 @@ def submit_backfill_runs(
         external_job = code_location.get_external_job(pipeline_selector)
     else:
         external_job = external_repo.get_full_external_job(external_partition_set.job_name)
-    for partition_data in result.partition_data:
+
+    partition_data_target = check.is_list(
+        [partition_names_or_ranges[0].start]
+        if isinstance(partition_names_or_ranges[0], PartitionKeyRange)
+        else partition_names_or_ranges,
+        of_type=str,
+    )
+    partition_set_execution_data = code_location.get_external_partition_set_execution_param_data(
+        external_repo.handle,
+        partition_set_name,
+        partition_data_target,
+        instance,
+    )
+    assert isinstance(partition_set_execution_data, ExternalPartitionSetExecutionParamData)
+
+    # Partition-scoped run config is prohibited at the definitions level for a jobs that materialize
+    # ranges, so we can assume that all partition data will have the same run config and tags as the
+    # first partition.
+    tags_by_key_or_range: Mapping[Union[str, PartitionKeyRange], Mapping[str, str]]
+    run_config_by_key_or_range: Mapping[Union[str, PartitionKeyRange], Mapping[str, Any]]
```

Comment:
At this point coming up with a name for `Union[str, PartitionKeyRange]` and using it in this code would be very helpful.

Alternatively you could canonicalize to `PartitionKeyRange`, which is probably preferable.

Reply:
You could have to have a class that "remembers" its lineage to create the correct tags (and maybe we should consider canonicalizing that as well?).

Reply:
>Alternatively you could canonicalize to PartitionKeyRange, which is probably preferable.

Yes, my plan was to normalize the entire job backfill pathway to `PartitionKeyRange` in a follow-up as a starting point for this workstream, since it is a relatively "contained" body of code that I now understand very well.